storage/modules/block_storage.go

```diff
@@ -1084,6 +1136,108 @@ func (b *BlockStorage) FindTransaction(
 	return newestBlock, newestTransaction, nil
 }
 
+func (b *BlockStorage) FindRelatedTransactions(
+	ctx context.Context,
+	transactionIdentifier *types.TransactionIdentifier,
+	db database.Transaction,
+) (*types.BlockIdentifier, *types.Transaction, []*types.Transaction, error) {
+	rootBlock, tx, err := b.FindTransaction(ctx, transactionIdentifier, db)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if rootBlock == nil {
+		return nil, nil, nil, nil
+	}
+
+	childIds, err := b.getChildren(ctx, tx, db)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// create map of seen transactions to avoid duplicates
+	seen := make(map[string]bool)
```

Comment:
usually we just do `map[string]struct{}` to create a "set" so that we don't need to explicitly set a value bc it will always be true